common/djangoapps/third_party_auth/tests/test_middleware.py

```diff
@@ -0,0 +1,31 @@
+"""
+Tests for third party auth middleware
+"""
+import mock
+from third_party_auth.tests.testutil import TestCase
+from third_party_auth.middleware import ExceptionMiddleware
+from requests.exceptions import HTTPError
+from django.test.client import RequestFactory
+from django.contrib.messages.middleware import MessageMiddleware
+
+
+class ThirdPartyAuthMiddlewareTestCase(TestCase):
+
+    @mock.patch('django.conf.settings.MESSAGE_STORAGE', 'django.contrib.messages.storage.cookie.CookieStorage')
+    def test_http_exception_redirection(self):
+        """
+        Test ExceptionMiddleware is correctly redirected to login page
+        when PSA raises HttpError exception.
+        """
+
+        test_url = '/login'
+        request = RequestFactory().get("dummy_url")
+        exception = HTTPError()
+        # Add error message for error in auth pipeline
+        MessageMiddleware().process_request(request)
+        response = ExceptionMiddleware().process_exception(
+            request, exception
+        )
```

Comment:
avoid using exception or any other python/django related key words as variable name. Also you don't need to put HTTPError() in a variable as it is not used else where.

Reply:
agree with @ahsan-ul-haq no need to put `HTTPError` in a variable. We can use it directly.